site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/controller/SiteController.java

```diff
@@ -35,6 +36,7 @@ public SiteController(SiteService siteService) {
         return ResponseEntity.status(HttpStatus.CREATED).body(new Response(siteId));
     }
 
+    @PostAuthorize("@authorisationService.filterUserSites(returnObject)")
```

Comment:
One thing to consider - you can use getBody in the PostAuthorize annotation so that the service filterUserSites method is not tied to ResponseEntity I.e.

```
@PostAuthorize("@authorisationService.filterUserSites(returnObject.getBody()")
```

And update the signature of filterUserSites. getBody returns a reference to the list, not a copy, so you can still filter it.